ui/v2.5/src/components/Shared/Select.tsx

```diff
@@ -454,22 +475,105 @@ export const MovieSelect: React.FC<IFilterProps> = (props) => {
 };
 
 export const TagSelect: React.FC<IFilterProps> = (props) => {
+  const [tagAliases, setTagAliases] = useState<Record<string, string[]>>({});
+  const [allAliases, setAllAliases] = useState<string[]>([]);
   const { data, loading } = useAllTagsForFilter();
-  const [createTag] = useTagCreate({ name: "" });
+  const [createTag] = useTagCreate();
   const placeholder = props.noSelectionString ?? "Select tags...";
 
-  const tags = data?.allTags ?? [];
+  const tags = useMemo(() => data?.allTags ?? [], [data?.allTags]);
+
+  useEffect(() => {
+    // build the tag aliases map
+    const newAliases: Record<string, string[]> = {};
+    const newAll: string[] = [];
+    tags.forEach((t) => {
+      newAliases[t.id] = t.aliases;
+      newAll.push(...t.aliases);
+    });
+    setTagAliases(newAliases);
+    setAllAliases(newAll);
+  }, [tags]);
+
+  const TagOption: React.FC<OptionProps<Option, boolean>> = (optionProps) => {
+    const { inputValue } = optionProps.selectProps;
+
+    let thisOptionProps = optionProps;
+    if (inputValue && !optionProps.label.includes(inputValue)) {
```

Comment:
`.includes()` does case sensitive matching. This means that typing in an option in lowercase while the actual option has (some) uppercase letter(s), or the other way around, it won't match and it is assumed to be an alias. In other words: this is the explanation of @bnkai's comment about `(alias)` being added randomly. So both should be converted to either lower or uppercase to make the matching case insensitive (or use some other matching method which isn't case sensitive).